ts/Extensions/Annotations/Popup.ts

```diff
@@ -369,35 +404,39 @@ H.Popup.prototype = {
         return button;
     },
     /**
-     * Get values from all inputs and create JSON.
+     * Get values from all inputs and selections then create JSON.
+     *
      * @private
-     * @param {Highcharts.HTMLDOMElement} - container where inputs are created
-     * @param {string} - add | edit | remove
-     * @return {Highcharts.PopupFieldsObject} - fields
+     *
+     * @param {Highcharts.HTMLDOMElement} parentDiv
+     *        The container where inputs and selections are created.
+     *
+     * @param {string} type
+     *         Type of the popup bookmark (add|edit|remove).
+     *
+     * @return {Highcharts.PopupFieldsObject}
      */
     getFields: function (
         parentDiv: HTMLDOMElement,
         type: string
     ): Highcharts.PopupFieldsObject {
-
-        let inputList = parentDiv.querySelectorAll('input'),
+        const inputList = parentDiv.querySelectorAll(INPUT),
+            selectList = parentDiv.querySelectorAll(SELECT),
             optionSeries = '#' + PREFIX + 'select-series > option:checked',
             optionVolume = '#' + PREFIX + 'select-volume > option:checked',
             linkedTo = parentDiv.querySelectorAll(optionSeries)[0],
-            volumeTo = parentDiv.querySelectorAll(optionVolume)[0],
-            seriesId,
-            param,
-            fieldsOutput: Highcharts.PopupFieldsObject;
+            volumeTo = parentDiv.querySelectorAll(optionVolume)[0];
+        let fieldsOutput: Highcharts.PopupFieldsObject;
 
         fieldsOutput = {
             actionType: type,
-            linkedTo: linkedTo && linkedTo.getAttribute('value'),
+            linkedTo: linkedTo && linkedTo.getAttribute('value') || '',
             fields: { }
-        } as any;
+        };
 
-        [].forEach.call(inputList, function (input: HTMLInputElement): void {
-            param = input.getAttribute(PREFIX + 'data-name');
-            seriesId = input.getAttribute(PREFIX + 'data-series-id');
+        inputList.forEach(function (input: HTMLInputElement): void {
```

Comment:
This won't work in IE

Reply:
IMHO I still prefer calling `forEach` directly on a converted array. To convert nodelist to array in old IE you can use `slice` like this:
```js
const inputList = Array.prototype.slice.call(parentDiv.querySelectorAll(INPUT));

inputList.forEach(function (...
...
```

Reply:
Is there any difference between `[].forEach.call(inputList, ...)` and `Array.prototype.slice.call(parentDiv.querySelectorAll(INPUT)).forEach(...)`?

Reply:
Apart from that it's look clearer (at least for me) it's the same.